From 6a7b4713249dd75f23f4b6ec9360cefda5a51d83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Juan=20Manuel=20L=C3=B3pez=20Carrillo?=
 <juanmanuellopezcarrillo@gmail.com>
Date: Tue, 14 Jul 2026 02:13:25 +0200
Subject: [PATCH 067/143] arm64: dts: sun55i-a523: wire up the CPU clocks and
 rails

cpu0-3 (little cluster) -> cpu_ccu CLK_CPUL + reg_dcdc1 (AXP717, vdd-cpul);
cpu4-7 (big cluster) -> cpu_ccu CLK_CPUB + reg_dcdc1_323 (AXP323, vdd-cpub).
With clock-latency-ns. No operating-points yet, so cpufreq does not start yet
(inert checkpoint: infrastructure ready, no live DVFS).
---
 .../arm64/boot/dts/allwinner/sun55i-a523.dtsi | 17 ++++++++++
 .../dts/allwinner/sun55i-t527-orangepi-4a.dts | 34 +++++++++++++++++++
 2 files changed, 51 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi b/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
index 1b195f564..11ac6c155 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun55i-a523.dtsi
@@ -7,6 +7,7 @@
 #include <dt-bindings/clock/sun8i-tcon-top.h>
 #include <dt-bindings/reset/sun8i-de2.h>
 #include <dt-bindings/clock/sun55i-a523-ccu.h>
+#include <dt-bindings/clock/sun55i-a523-cpu-ccu.h>
 #include <dt-bindings/clock/sun55i-a523-mcu-ccu.h>
 #include <dt-bindings/clock/sun55i-a523-r-ccu.h>
 #include <dt-bindings/reset/sun55i-a523-ccu.h>
@@ -35,6 +36,8 @@ cpu0: cpu@0 {
 			device_type = "cpu";
 			reg = <0x000>;
 			enable-method = "psci";
+			clocks = <&cpu_ccu CLK_CPUL>;
+			clock-latency-ns = <244144>;
 		};
 
 		cpu1: cpu@100 {
@@ -42,6 +45,8 @@ cpu1: cpu@100 {
 			device_type = "cpu";
 			reg = <0x100>;
 			enable-method = "psci";
+			clocks = <&cpu_ccu CLK_CPUL>;
+			clock-latency-ns = <244144>;
 		};
 
 		cpu2: cpu@200 {
@@ -49,6 +54,8 @@ cpu2: cpu@200 {
 			device_type = "cpu";
 			reg = <0x200>;
 			enable-method = "psci";
+			clocks = <&cpu_ccu CLK_CPUL>;
+			clock-latency-ns = <244144>;
 		};
 
 		cpu3: cpu@300 {
@@ -56,6 +63,8 @@ cpu3: cpu@300 {
 			device_type = "cpu";
 			reg = <0x300>;
 			enable-method = "psci";
+			clocks = <&cpu_ccu CLK_CPUL>;
+			clock-latency-ns = <244144>;
 		};
 
 		cpu4: cpu@400 {
@@ -63,6 +72,8 @@ cpu4: cpu@400 {
 			device_type = "cpu";
 			reg = <0x400>;
 			enable-method = "psci";
+			clocks = <&cpu_ccu CLK_CPUB>;
+			clock-latency-ns = <244144>;
 		};
 
 		cpu5: cpu@500 {
@@ -70,6 +81,8 @@ cpu5: cpu@500 {
 			device_type = "cpu";
 			reg = <0x500>;
 			enable-method = "psci";
+			clocks = <&cpu_ccu CLK_CPUB>;
+			clock-latency-ns = <244144>;
 		};
 
 		cpu6: cpu@600 {
@@ -77,6 +90,8 @@ cpu6: cpu@600 {
 			device_type = "cpu";
 			reg = <0x600>;
 			enable-method = "psci";
+			clocks = <&cpu_ccu CLK_CPUB>;
+			clock-latency-ns = <244144>;
 		};
 
 		cpu7: cpu@700 {
@@ -84,6 +99,8 @@ cpu7: cpu@700 {
 			device_type = "cpu";
 			reg = <0x700>;
 			enable-method = "psci";
+			clocks = <&cpu_ccu CLK_CPUB>;
+			clock-latency-ns = <244144>;
 		};
 	};
 
diff --git a/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts b/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
index 53e1038f9..91ae0a85a 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
+++ b/arch/arm64/boot/dts/allwinner/sun55i-t527-orangepi-4a.dts
@@ -233,6 +233,40 @@ reg_vcc5v: regulator-vcc5v {
 	};
 };
 
+/* Little cluster (cpu0-3) is powered by the AXP717 DCDC1 (vdd-cpul). */
+&cpu0 {
+	cpu-supply = <&reg_dcdc1>;
+};
+
+&cpu1 {
+	cpu-supply = <&reg_dcdc1>;
+};
+
+&cpu2 {
+	cpu-supply = <&reg_dcdc1>;
+};
+
+&cpu3 {
+	cpu-supply = <&reg_dcdc1>;
+};
+
+/* Big cluster (cpu4-7) is powered by the AXP323 DCDC1 (vdd-cpub). */
+&cpu4 {
+	cpu-supply = <&reg_dcdc1_323>;
+};
+
+&cpu5 {
+	cpu-supply = <&reg_dcdc1_323>;
+};
+
+&cpu6 {
+	cpu-supply = <&reg_dcdc1_323>;
+};
+
+&cpu7 {
+	cpu-supply = <&reg_dcdc1_323>;
+};
+
 &ehci0 {
 	status = "okay";
 };
-- 
2.53.0

